test/common/test_runner.go

```diff
@@ -133,6 +135,55 @@ func CreateNamespaceIfNeeded(t *testing.T, client *Client, namespace string) {
 		if err != nil {
 			t.Fatalf("The default ServiceAccount was not created for the Namespace: %s", namespace)
 		}
+
+		// If the "default" Namespace has a secret called
+		// "kn-eventing-test-pull-secret" then use that as the ImagePullSecret
+		// on the "default" ServiceAccount in this new Namespace.
+		// This is needed for cases where the images are in a private registry.
+
+		// Get the Interfaces we need to access the resources in the cluster
+		defSecI := client.Kube.Kube.CoreV1().Secrets("default")
+		nsSAI := client.Kube.Kube.CoreV1().ServiceAccounts(namespace)
+		nsSecI := client.Kube.Kube.CoreV1().Secrets(namespace)
+
+		testSecret, _ := defSecI.Get(TestPullSecretName, metav1.GetOptions{})
+
+		// Check again. I've seen cases where it lies and if we need it
+		// then the test will fail w/o it, so check again just to be sure.
+		if testSecret == nil {
+			testSecret, _ = defSecI.Get(TestPullSecretName, metav1.GetOptions{})
+		}
+
+		if testSecret != nil {
```

Comment:
This function seems similar in description, yet it's different and I'm trying to understand why? :) Can this be hoisted into a separate function and reused instead of having two versions of it.

Reply:
they are similar but different. This one only deals with the "default" SA and knows it can blindly copy the secret. Th other one deals with a new SA (so not "default") and needs to check to see if the secret already exists so it doesn't try to duplicate it.

Reply:
They both have the check and re-check (which I must say is extremely worrying and seems extremely flaky and a bug in the k8s if that's the case).
I think having a method that deals with copying the secret into the namespace would be good to hoist out, then rejiggering the ServiceAccount secret could be the only difference between the two if I understand correctly?